### ISSUE TITLE
vnodeq histogram disappear when the number of vnodes is 2 on a node. [JIRA: RIAK-2357]

### ISSUE THREAD

**Issue by ksauzz** [basho/riak_core#804]
#### Phenomenon

Problem: `riak_[kv|pipe]_vnodeq_[max|mean|min|median|total]` **disappears** on a node which **own 2 vnodes**.

```
% riak-admin status | grep -e running -e vnodeq -e ring_creation_size -e ownership
riak_kv_vnodes_running : 2
riak_pipe_vnodes_running : 2
ring_creation_size : 8
ring_ownership : <<"[{'riak5@127.0.0.1',1},\n {'riak4@127.0.0.1',1},\n {'riak3@127.0.0.1',2},\n {'riak2@127.0.0.1',2},\n {'riak1@127.0.0.1',2}]">>
```

Expected behavior: `riak_[kv|pipe]_vnodeq_[max|mean|min|median|total]` **appears** on a node which **doesn't own 2 vnode**.

```
% riak-admin status | grep -e running -e vnodeq -e ring_creation_size -e ownership
riak_kv_vnodeq_max : 0
riak_kv_vnodeq_mean : 0.0
riak_kv_vnodeq_median : 0
riak_kv_vnodeq_min : 0
riak_kv_vnodeq_total : 0
riak_kv_vnodes_running : 1
riak_pipe_vnodeq_max : 0
riak_pipe_vnodeq_mean : 0.0
riak_pipe_vnodeq_median : 0
riak_pipe_vnodeq_min : 0
riak_pipe_vnodeq_total : 0
riak_pipe_vnodes_running : 1
ring_creation_size : 8
ring_ownership : <<"[{'riak5@127.0.0.1',1},\n {'riak4@127.0.0.1',1},\n {'riak3@127.0.0.1',2},\n {'riak2@127.0.0.1',2},\n {'riak1@127.0.0.1',2}]">>
```
#### Reproduction
1. prepare 5 nodes with ring size = 8
2. make a cluster by them
3. wait for finishing handoffs.
4. check http stats or `riak-admin status`, them you can see the problem above.
#### Root cause

`exometer_util:histogram/1` doesn't return a value as a consistent form. It can return empty list which makes vnodeq histogram disappear even though a caller expects a value as proplist of histogram.

A vnodeq histogram depends on the form of a return value of `exometer_util:histogram/1`. That is a proplist like:

```
2> exometer_util:histogram([0]).
[{n,1},
 {mean,0.0},
 {total,0},
 {min,0},
 {50,0},
 {median,0},
 {75,0},
 {90,0},
 {95,0},
 {99,0},
 {999,0},
 {max,0}]
```

But `exometer_util:get_statistics2/4` which is called internally has [the special branch](https://github.com/basho/exometer_core/blob/1.0.0-basho3/src/exometer_util.erl#L239-L243). That returns `[]` when a value is `[0,0]` which is passed when a node runs 2 vnodes only.

```
3> exometer_util:histogram([0,0]).
[]
```

This causes the problem.

**Comment by kuenishi** [comment]
https://github.com/Feuerlabs/exometer_core/issues/67 might be one way to fix this.

**Comment by kuenishi** [comment]
I tried to open a PR at basho/exometer_core by pushing a branch to cherry-pick the fix https://github.com/Feuerlabs/exometer_core/pull/67, but I couldn't because it is not in the list of full-push-pull access group. Might it be a bigger challenge to update or to rebase exometer_core to the latest version and apply our changes?

**Comment by Basho-JIRA** [comment]
Opened PR https://github.com/basho/exometer_core/pull/10 to cherry pick Kota's patch 

***[posted via JIRA by Mark Allen]***

**Comment by Basho-JIRA** [comment]
Opened PR https://github.com/basho/exometer_core/pull/10 to cherry pick Kota's patch and https://github.com/basho/riak_core/pull/817 to pull in updated exometer tag 

**_[posted via JIRA by Mark Allen]**_

**Comment by Basho-JIRA** [comment]
Need to get the riak_core PR reviewed and merged. 

**_[posted via JIRA by Mark Allen]**_

**Comment by Basho-JIRA** [comment]
Mark's original PRs are all merged, and I've forward merged the fix from 2.0 to develop-2.2 in https://github.com/basho/riak_core/pull/822. Marked as fixed. 

**_[posted via JIRA by Nick Marino]**_
